### ISSUE TITLE
BigBox pause screen exiting emulator issue

### ISSUE THREAD

**Issue by laxdragon** [schellingb/dosbox-pure#65]
I've setup DOSBox Pure under LaunchBox/BigBox (Windows). I'm having an issue using the pause screen to exit out of the emulator, it does not actually exit, but instead returns to the emulator and mutes it (volume control appears onscreen). Attempting a second time unmutes. I can still keep a keyboard around and exit out with the ESC key, but I do like trying to do everything from a controller. Note, other RetroArch cores exit out just fine this way, but there is something specific about DOSBox Pure. Not sure if this issue is with the BB front end itself, but since this is the only core doing this, filing here anyway.

Steps to reproduce

1. Hit Xbox button on controller to bring up BigBox Pause menu
2. Choose Exit Emulator.

**Comment by laxdragon** [comment]
Got a tip from the LaunchBox forums that works around the issue.

Go to: Tools > Manage Emulators > RetroArch. Under the "Exit AutoHotKey Script" tab enter:

`WinClose, ahk_exe retroarch.exe`

**Comment by schellingb** [comment]
Is there a restart option in that pause menu you mention? Does it work? There is an exit option in the start menu (where one can select the program to launch), does that exit the emulator?

But unless someone can explain to me technically how "Exit Emulator" in that pause menu is different from pressing either "Close Content" or "Quit RetroArch" in the RetroArch menu (which works perfectly fine) I don't think I can help with this.

**Comment by Solo187** [comment]
I can confirm that it doesn't work when using an exit combination on gamepad either. I have select + x as the exit combination & with this core it just mutes the sound. As far as i know, exit just sends the exit key to the emulator. Which in this case would be esc. But obv that works on a keyboard so something extra is going on here.

**Comment by schellingb** [comment]
@Solo187 Are you talking about RetroArch or also about BigBox?
Do the another libretro DOSBox cores (svn or core) work correct in this regard?

**Comment by Solo187** [comment]
Yes sorry, this is also whilst using Bigbox. You can setup buttons on your gamepad to exit the emulator or bring up a pause screen. As far as i know, the way it exits the emulator is by sending the exit key to the emulator via a button combination on your gamepad. In retroarch case it would be the esc key. However, in the case whilst using your core, it only mutes the sound. I can only think that its something to do with how you simulate the keyboard using the pad input?

**Comment by schellingb** [comment]
So it just sends an ESC keypress to RetroArch? But doesn't it by default require two presses of the ESC key to quit RetroArch? Also... that seems like a horrible "solution".

**Comment by Solo187** [comment]
Yep, i think that's all it does. So i think it must be some kind of conflict with however your core handles the pad inputs mixed with keyboard inputs or something as obviously pressing esc on regular keyboard exits like normal.

**Comment by schellingb** [comment]
Why would it have anything to do with pad input or how the core handles input? If it's sending an ESC key and RetroArch has that key bound to the hotkey action 'Quit RetroArch' it should just work. This part of my first reply is still the most relevant.

> But unless someone can explain to me technically how "Exit Emulator" in that pause menu is different from pressing either "Close Content" or "Quit RetroArch" in the RetroArch menu (which works perfectly fine) I don't think I can help with this.

Also

> Do the another libretro DOSBox cores (svn or core) work correct in this regard?

**Comment by Solo187** [comment]
It was just a thought as it works as it should with every other core. It's no biggie to me. I noticed it wasn't working correctly but i just use the menus to exit with this particular core. Maybe the op could get some more info

**Comment by laxdragon** [comment]
Like I said in my original post, the BigBox Pause Screen "Exit" option works fine in all other cores I've tested. It only does this weird mute and not exit thing with DOSBox Pure. I've worked around it for now, by changing the macro used by the exit menu. I really don't know how the original option works, I would assume it is just passing the Esc key back to the emulator, but perhaps it is doing more than that.

**Comment by Solo187** [comment]
It is strange. I had a look & retroarch hotkey for mute is f9. It seems for some reason the exit command is sending that to the emulator instead of esc. Verified by changing the retroarch quit hotkey to f9
